crates/core/tedge_api/src/mqtt_topics.rs

```diff
@@ -91,8 +93,9 @@ impl MqttSchema {
     ///     "te/device/child001///a/sensors/meta"
     /// );
     /// ```
+    // TODO: add test
     pub fn topic_for(&self, entity: &EntityTopicId, channel: &Channel) -> mqtt_channel::Topic {
-        let topic = format!("{}/{}/{}", self.root, entity, channel);
+        let topic = format!("{}/{}{}", self.root, entity, channel);
```

Comment:
What's the purpose of this change? It sounds weird to me to force a channel name to include a `/` prefix.

Reply:
[It's because the registration message topic is e.g. `te/device/main/service/servicename`, instead of `te/device/main/service/servicename/`](https://thin-edge.github.io/thin-edge.io/next/references/mqtt-api/#entity-registration). These are two different topics in MQTT and the entity store would not pick up the registration message when the topic had a trailing slash. I was rushing a bit and didn't add a test case, but I'll note it down as a TODO.

Reply:
Thanks for having updated this. The new version wraps the subtlety inside the `topic_for` function instead of exposing it to all uses of `target.to_string()`.